basic_auth.go

```diff
@@ -21,8 +21,7 @@ type basicAuth struct {
 // default error handler if you wish to serve a custom template/response.
 type AuthOptions struct {
 	Realm               string
-	User                string
-	Password            string
+	AuthFunc            func(string, string) bool
```

Comment:
Yes I know that was my concern too, however probably most of the users use the wrapper anyway SimpleBasicAuth. One option would be to just add the AuthFunc as an optional feature. What do you think ?